invenio_search/utils.py

```diff
@@ -10,28 +10,38 @@
 
 import os
 
+from flask import current_app
 
-def build_index_name(*parts):
+
+def prefix_index(app, index):
+    """Prefixes the given index if needed.
+
+    :param app: Flask app to get the config from.
+    :param index: Name of the index to prefix.
+    :returns: A string with the new index name prefixed in needed.
+    """
+    index_prefix = app.config['SEARCH_INDEX_PREFIX']
+    if index.startswith(index_prefix):
```

Comment:
Which cases is this covering? Is it only when ``SEARCH_INDEX_PREFIX == ''`` or are there other cases?

Example: If you decided to name your prefix ``records`` and your index starts with ``records`` I would still expect the final index name to be ``records-records`` (might not make much sense, but it's consistent 😄).

Reply:
That is there because the libs at some point start doing nested calls passing the index string around, so if you have to make sure to prepend the perfix only once. Unfortunately, all the info you get is the index name string, so you can't add random info (like a boolean property `prefixed`). But, if you want your indices to end up as `record-record` the prefix should be `record-`, with the last dash, and that will not really collide (unless you want `record-record-something`). But I don't know how to prevent that from happening without changing substantially the code.

Reply:
Got it. Yeah, not much to do about it. Perhaps just add a quick comment in the code, that this line is there due to nested calls.